code-challenges.test.js

```diff
@@ -36,21 +55,67 @@ const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
 
 
+describe('onlyNums', () => {
+    it('Create a function that takes in an array and returns a new array of only odd numbers sorted from least to greatest.', () => {
+        const fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
+        // Expected output: [-9, 7, 9, 199]
+
+        const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
+        // Expected output: [-823, 7, 23]
+
+        expect(onlyNums(fullArr1)).toEqual([-9, 7, 9, 199])
+        expect(onlyNums(fullArr2)).toEqual([-823, 7, 23])
+    })
+})
+
+
 // b) Create the function that makes the test pass.
 
+// sort will sort the data but it wont completely work on numbers without the a-b addition (which I have to lookup)
+// create a local variable that filters through the array assessing each value
+// the return needs to be only numbers and (&&) only those numbers that have a remainder after being devided by two
+
+const onlyNums = (array) => {
+    let dataType = array.sort(function (a, b) { return a - b }).filter(value => {
+        return typeof (value) === "number" && (value) % 2 !== 0
+    })
+    return dataType
+}
+
 
 // --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const numbersToAdd1 = [2, 4, 45, 9]
-// Excpected output: [2, 6, 51, 60]
+describe('addingArray', () => {
+    it('Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.', () => {
+        const numbersToAdd1 = [2, 4, 45, 9]
+        // Excpected output: [2, 6, 51, 60]
+
+        const numbersToAdd2 = [0, 7, -8, 12]
+        // Expected output: [0, 7, -1, 11]
+
+        const numbersToAdd3 = []
+        // Expected output: []
 
-const numbersToAdd2 = [0, 7, -8, 12]
-// Expected output: [0, 7, -1, 11]
+        expect(addingArray(numbersToAdd1)).toEqual([2, 6, 51, 60])
+        expect(addingArray(numbersToAdd2)).toEqual([0, 7, -1, 11])
+        expect(addingArray(numbersToAdd3)).toEqual([])
+    })
+})
 
-const numbersToAdd3 = []
-// Expected output: []
 
 
 // b) Create the function that makes the test pass.
+// createing a for loop that taking in an array and returns a new arrway with the vaule of each index location added of the old array added to the previous index of the new array this should result in the numbers of the new array effetively accumulating 
+// need to make sure that if there are no values no values are returned
+// I can use .length to ensure that the correct index values are getting added together and .push to return the correct new values to the new array
+
+function addingArray(array) {
+    let empty = []
+    for (let num of array) {
+        if (!empty.length) empty.push(Number(num))
+        else empty.push(empty[empty.length - 1] + Number(num))
+    }
+    return empty
+}
```

Comment:
This approach certainly does work, can you think of how you could write this with a higher-order function?  Also would love to hear your feedback on why you chose function notation instead of arrow notation.